File: src/main/java/com/example/ptiringo/domain/entity/Downtown.java
package com.example.ptiringo.domain.entity;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

import lombok.Data;

/**
 * 商業集積地区
 * @author ptiringo
 */
@Data
@Entity
public class Downtown {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	/** 都道府県コード */
	private Integer prefectureCode;
	/** 商業集積地番号1 */
	private Integer code1;
	/** 商業集積地番号2 */
	private Integer code2;
	/** 商店街名 */
	private String name;
	/** 集積細分コード */
	private Integer commercialAccumulationCode;
	/** 事業所数 */
	private Integer officeCount;
	/** 大規模小売店舗内事業所数 */
	@Column(nullable = true)
	private Integer insideShopCount;
	/** 従業者数 */
	private Integer employeeCount;
	/** 年間商品販売額 */
	private Long annualSalesTurnover;
	/** 売場面積 */
	@Column(nullable = true)
	private Integer area;

	/** 緯度 */
	@Column(nullable = true)
	private Double latitude;
	/** 経度 */
	@Column(nullable = true)
	private Double longitude;
}

File: src/main/kotlin/com/example/ptiringo/domain/entity/Downtown.kt
package com.example.ptiringo.domain.entity

import javax.persistence.*

/**
 * 商業集積地区
 *
 * @property id ID
 * @property prefecture 都道府県
 * @property code1 商業集積地番号1
 * @property code2 商業集積地番号2
 * @property name 商店街名
 * @property commercialAccumulationCode 集積細分コード
 * @property officeCount 事業所数
 * @property insideShopCount 大規模小売店舗内事業所数
 * @property employeeCount 従業者数
 * @property annualSalesTurnover 年間商品販売額
 * @property area 売場面積
 * @property point 位置
 * @author ptiringo
 */
@Entity
data class Downtown(
        @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
        val id: Long,

        @Embedded
        val prefecture: Prefecture,

        val code1: Int,

        val code2: Int,

        val name: String,

        val commercialAccumulationCode: Int,

        val officeCount: Int,

        @Column(nullable = true)
        var insideShopCount: Int?,

        val employeeCount: Int,

        val annualSalesTurnover: Long,

        @Column(nullable = true)
        var area: Int?,

        @Embedded
        val point: Point
)